picard/track.py

```diff
@@ -368,6 +374,18 @@ def _convert_folksonomy_tags_to_genre(self):
             join_with=config.setting['join_genres']
         )
 
+    def add_folksonomy_tags(self):
+        config = get_config()
+        tags = Counter(self._folksonomy_tags)
+        tags += self.album._folksonomy_tags
+        self.metadata['_folksonomy_tags'] = self._genres_to_metadata(tags, join_with=config.setting['join_genres'])
```

Comment:
To my understanding we can do this simpler and don't need to filter the values for `_folksonomy_tags` and `_genres`. We just include all the loaded values in either variable. Only the final `genre` variable gets filtered.

But I asked UltimateRiff on PICARD-2607 for clarification also.

What I'm also unsure about if `_folksonomy_tags` should strictly only contain the non-genre tags. In that case we would filter out everything that is in `_genres`.

Reply:
Hmm I think _folksonomy_tags should only have the non-genre tags. But well I'll just to see what UltimateRiff expects before I implement it.

For the implementation though, is iterating through _genres.keys() and deleting them from _folksonomy_tags going to be good enough or is there a more efficient method?

Reply:
@phw [UltimateRiff replied](https://tickets.metabrainz.org/browse/PICARD-2607?focusedId=73844&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-73844):
> on the first expectation, I'm honestly not sure what my preference would be
on the second tho, I think I'd expect _folksonomy_tags to be only non-genre tags


I think for first, including everything is the right way to go for now.    
For the deleting genres from folksonomy tags, is my earlier approach fine? 
> Is iterating through _genres.keys() and deleting them from _folksonomy_tags going to be good enough or is there a more efficient method?

Reply:
Implemented deleting tags from genres in commit: 58286b778343fd23f02df682ca3ee2eea42044c6